### ISSUE TITLE
Void Overflow not working as expected/wanted

### ISSUE THREAD

**Issue by radzo73** [P3pp3rF1y/SophisticatedStorage#67]
Given a storage that is full (either with a lot of items, or all the slots remember their contents), and an Advanced Void Upgrade set to Void Overflow (Allows single-slot items to be filled with the item and voids anything _that overflows_); when I insert an item that cannot go into a storage (for example, I use a Hopper to insert Nether Stars into a storage full of Cobblestone)

**Expected Behavior:** These Nether Stars do not go into the storage (as there is no room for them), and stay in the Hopper.

**Actual Behavior:** These Nether Stars _do_ go into the storage, but get voided as there is no room for them.

This is a problem because I would like to incorporate my mob farm storage into my Refined Storage, but I can't because:
-If the drives are prioritized, then unused mob drops will inevitably fill up my drives.
-If the mob farm storage is prioritized, then everything imported into the RS system will go to the mob farm storage, and get voided.

TL;DR, if an item tries to go into a storage that has an Advanced Void Upgrade set to Void Overflow but cannot because no slots in the storage can accept the incoming item, then it should stay in the original inventory/buffer, instead of being voided.

**Comment by P3pp3rF1y** [comment]
Your expected behavior is exactly how it works for me so I will need more info on how to recreate / how you have set this up to be able to recreate. I have also made some adjustments to void upgrade several months so perhaps you're using an old version (can't tell as none was shared)

**Comment by radzo73** [comment]
using `sophisticatedstorage-1.18.2-0.5.1.107.jar` (latest is `sophisticatedstorage-1.18.2-0.5.9.130.jar`, which does not affect the Advanced Void Upgrade)

Here's my setup; a Netherite Barrel with an Advanced Void Upgrade set to Void Overflow, and all slots remembering their contents.
![image](https://user-images.githubusercontent.com/26673668/187324517-1dee90d9-9722-4765-82e8-e56303d04e50.png)

I set all the slots to remember their contents because I didn't want a bunch of other stuff filling this barrel, like sticks, or Nether Stars. After all, if all the slots remember their contents, then nothing else should enter, and nothing else should be voided, right?
Let's hopper in a stack of sticks (which none of the slots will accept), and see what happens.
![image](https://user-images.githubusercontent.com/26673668/187325436-060213d7-08c5-435f-af37-1620c55106cb.png)
aaaaaaand they're getting voided.

My guess is that the Void Upgrade (Void Overflow) takes priority over slot memories; where inserted items that cannot be stored get voided.
What I would prefer to happen is that the slot memories take priority over the Void Upgrade (Void Overflow); where inserted items that cannot be stored are not accepted, and instead stay in the inserting inventory.

It's kinda like if I had a 2x2 Storage Drawer with a Void Upgrade, and I tried to hopper in a stack of sticks (except all slots on this Drawer are full with, say, saplings), then the drawer would not accept the sticks (as there is no room for them), instead of trying to accept them, but realizing there is no room for them, and then just voiding them.

**Comment by P3pp3rF1y** [comment]
right, I wasn't able to recreate with vanilla hopper before but was able to with simple storage so I have the issue fixed in dev now

**Comment by P3pp3rF1y** [comment]
Already released for some time. Closing...
